### PR TITLE
chore: update BE dependencies

### PR DESCRIPTION
**Pull request by atarashansky**
# Reason for change
 - #539 
 - Dockerfile was using `hosted/requirements.txt` which are very outdated and vestigial from old configuration. This file was removed and replaced with an upgraded `server/requirements.txt`.

# Testing
 - Confirmed working with unit and smoke tests, and tested in [rdev stack](https://atar-be-update-explorer.rdev.single-cell.czi.technology/e/00bdbc57-1699-4ac4-8a10-cb7fb25cb1ad.cxg/).

### CODE REVIEW

File: .github/workflows/push_tests.yml [hunk 1]
```diff
@@ -19,10 +19,10 @@ jobs:
       - uses: actions/checkout@v2
       - run: |
           git fetch --depth=1 origin +${{github.base_ref}}
-      - name: Set up Python 3.7
+      - name: Set up Python 3.8
```
Comment (atolopko-czi) [inline .github/workflows/push_tests.yml:22]: good reason not to upgrade higher?

Comment (atarashansky) [reply .github/workflows/push_tests.yml:22]: Yes exactly! matching docker python version.

File: server/requirements.txt [hunk 2]
```diff
@@ -1,24 +1,23 @@
...
-bitarray>=2.3.6
-boto3>=1.12.18
-envyaml>=1.9
-Flask~=2.2
-Flask-Compress>=1.4.0
-Flask-Cors>=3.0.9  # CVE-2020-25032
-Flask-RESTful>=0.3.9
-flask-server-timing>=0.1.2
-flask-talisman>=0.7.0
+anndata>=0.8.0
```
[the remark this answers is not in the document]
Comment (atolopko-czi) [reply server/requirements.txt:1]: just anndata; it's the only remaining range-based version requirement, if my eyes do not deceive me. :)

[the remark this answers is not in the document]
Comment (atarashansky) [reply server/requirements.txt:1]: lol, i think it was my memory that deceived me!

**Review by atolopko-czi — APPROVED** [review]

**Pull request merged**